createServer updates

bnjmnt4n opened the pull request:
h5bp.createServer updates to optimize compress options filtering

ngryman commented on README.md:
```diff
@@ -147,7 +147,7 @@ Tells if you want to log server requests or not. This can also be an object cont
 ### compress
 
-Tells if you want to serve `gzipped` content or not. By default this is `true`.
+Tells if you want to serve `gzipped` content or not. By default this is `true`. Alternatively, you can pas in an object with options passed into the `compress` middleware.
```
Typo & not very explicit.
Could you rephrase it?

ngryman commented on lib/h5bp.js:
```diff
@@ -113,7 +113,7 @@ h5bp = module.exports = function(options) {
  * Adds and configures static and favicon middlewares to serve static files.
  *
- * @param {Boolean|Object} [options.logger=false] true or false or logger options.
+ * @param {Boolean|Object|Function} [options.logger=false] true or false or logger options.
```
Didn't you mean `compress` param instead of `logger`? Or both?

ngryman commented on test/h5bp.js:
```diff
@@ -381,13 +381,17 @@ describe('h5bp', function() {
-				.expect('Vary', 'Accept-Encoding')
-				.expect('Content-Encoding', 'gzip')
-				.expect(200, done);
+		describe('compressed files', function() {
```
Could you also replicate this test in the `standalone` section?

ngryman commented on lib/h5bp.js:
```diff
@@ -137,12 +137,27 @@ h5bp.createServer = function(options, callback) {
+		if (!compress.filter) {
+			compress.filter = function(req, res) {
+				var header = res.getHeader('Content-Type');
+				var test = /(?:text\/.*|application\/(?:j(?:son|avascript)|(?:(?:xhtml|rss|atom)\+)?xml|vnd\.ms-object|x-font-ttf)|image\/svg\+xml|font\/opentype)/.test(header);
```
Great! It hads a lot more content types :)
Have you taken it from the Apache version?

bnjmnt4n committed:
`h5bp.createServer` updates

Optimize `compress` option, allow `logger` to be a function, and update readme and tests
